blueprint-manager/src/main/java/com/sequenceiq/cloudbreak/blueprint/smartsense/SmartSenseConfigProvider.java

```diff
@@ -88,7 +90,7 @@ public boolean additionalCriteria(BlueprintPreparationObject source, String blue
     private String addSmartSenseServerToBp(BlueprintTextProcessor blueprintProcessor, Iterable<HostgroupView> hostgroupViews,
             Collection<String> hostGroupNames) {
         if (!blueprintProcessor.componentExistsInBlueprint(HST_SERVER_COMPONENT)) {
-            String aHostGroupName = hostGroupNames.stream().findFirst().get();
+            String aHostGroupName = MASTER_HOST_GROUP_NAME;
```

Comment:
I don't see the relation with flow retry. Could you please explain? Otherwise what is the guarantee that the name of the master host group is master?

Reply:
+1 it should not be hard coded, it can be anything

Reply:
I was not sure about this exact change either. Something had to be changed though, since the hostGroupNames is a set, the order of the entries can change. A module test is randomly failing because of this line, and I am also not sure if it is okay that a random host group is elected as server.

Reply:
Oh, wow. @biharitomi do you know what should be the expected behavior here?

Reply:
CentralBlueprintUpdaterRollingtest is the twst in question btw.